src/web/utils/cursorUtils.ts

```diff
@@ -54,28 +54,25 @@ function setCursorPosition(target: MarkdownTextInputElement, startIndex: number,
 
 function scrollIntoView(target: MarkdownTextInputElement, node: TreeNode) {
   const targetElement = target;
-  if (node.type === 'br' && node.parentNode?.parentNode?.type === 'line') {
-    // If the node is a line break, scroll to the parent paragraph, because Safari doesn't support scrollIntoView on br elements
-    node.parentNode.parentNode.element.scrollIntoView({
-      block: 'nearest',
-    });
-  } else {
-    const selection = window.getSelection();
-    if (selection && selection.rangeCount > 0) {
-      const range = selection.getRangeAt(0);
-      const caretRect = range.getBoundingClientRect();
-      const targetRect = target.getBoundingClientRect();
-
-      // In case the caret is below the visible input area, scroll to the end of the node
-      if (caretRect.top + caretRect.height > targetRect.top + targetRect.height) {
-        targetElement.scrollTop = caretRect.top + caretRect.height - targetRect.top - targetRect.height + target.scrollTop;
-      }
-    }
+  const orderIndex = Number(node.orderIndex.split(',')[0]);
+  const currentLine = target.tree.childNodes[orderIndex]?.element;
+  const scrollTargetElement = currentLine || node.element;
 
-    node.element.scrollIntoView({
-      block: 'nearest',
-    });
+  const selection = window.getSelection();
+  if (selection && selection.rangeCount > 0) {
```

Comment:
`selection` is no longer being used here. Do we still need this condition?

Reply:
Good catch, I forgot to remove it